certbot/main.py

```diff
@@ -382,6 +382,28 @@ def register(config, unused_plugins):
     # exist or not.
     account_storage = account.AccountFileStorage(config)
     accounts = account_storage.find_all()
+    reporter_util = zope.component.getUtility(interfaces.IReporter)
+    add_msg = lambda m: reporter_util.add_message(m, reporter_util.MEDIUM_PRIORITY)
+
+    if config.deactivate:
+        if len(accounts) == 0:
+            add_msg("Could not find existing account to deactivate.")
+        else:
+            yesno = zope.component.getUtility(interfaces.IDisplay).yesno
+            prompt = ("Are you SURE you would like to irrevocably deactivate "
+                      "your account? You will lose access to the domains "
+                      "associated with it.")
+            wants_deactivate = yesno(prompt, yes_label='Deactivate',
+                                     no_label='Abort', cli_flag='--deactivate',
+                                     default=False)
+            if wants_deactivate:
+                acc, acme = _determine_account(config)
+                acme_client = client.Client(config, acc, None, None, acme=acme)
+                acme_client.acme.deactivate(acc.regr)
+                add_msg("Account deactivated.")
+            else:
+                add_msg("Deactivation aborted.")
```

Comment:
Like above, I think we should just return this string here. The user requested action could not be completed and we effectively do this in [other places](https://github.com/certbot/certbot/blob/master/certbot/main.py#L187) if the user cancels their request.